pygfx/materials/_points.py

```diff
@@ -238,6 +238,118 @@ class PointsGaussianBlobMaterial(PointsMaterial):
     """
 
 
+class PointsMarkerMaterial(PointsMaterial):
+    """A material to render points as markers.
+
+    Markers come in a variety of shapes, and have an edge with a separate color.
+    """
+
+    uniform_type = dict(
+        PointsMaterial.uniform_type,
+        edge_color="4xf4",
+        edge_width="f4",
+    )
+
+    def __init__(self, *, marker="circle", edge_width=1, edge_color="black", **kwargs):
+        super().__init__(**kwargs)
+        self.marker = marker
+        self.edge_width = edge_width
+        self.edge_color = edge_color
+
+    @property
+    def edge_color(self):
+        """The color of the edge of the markers."""
+        return Color(self.uniform_buffer.data["edge_color"])
+
+    @edge_color.setter
+    def edge_color(self, edge_color):
+        edge_color = Color(edge_color)
+        self.uniform_buffer.data["edge_color"] = edge_color
+        self.uniform_buffer.update_range(0, 1)
+
+    @property
+    def edge_width(self):
+        """The width of the edge of the markers, in logical pixels (or world/model space if ``size_space``is set)."""
+        return float(self.uniform_buffer.data["edge_width"])
+
+    @edge_width.setter
+    def edge_width(self, edge_width):
+        self.uniform_buffer.data["edge_width"] = float(edge_width)
+        self.uniform_buffer.update_range(0, 1)
+
+    @property
+    def marker(self):
+        """The type/shape of the markers.
+
+        Possible values are:
+
+        * 'circle' or 'o"
+        * 'ring'
+        * 'square' or 's'
+        * 'diamond' or 'D'
+        * 'plus' or '+'
+        * 'cross' or 'x'
+        * 'asterix'
+        * 'triangle_up' or '^'
+        * 'triangle_down' or ''
+        * 'triangle_left' or '<'
+        * 'triangle_right' or '>'
+        * 'heart' or '♥'
+        * 'spade'
+        * 'club'
+        * 'pin'
+
+        """
+        # TODO: is marker a good name?
+        # Note: MPL calls this 'marker', Plotly calls this 'symbol'
+        return self._store.marker
+
+    @marker.setter
+    def marker(self, name):
+        # Define possible values, see:
+        # https://matplotlib.org/stable/api/markers_api.html
+        # https://plotly.com/python/marker-style/#custom-marker-symbols
+        possible_names = [
+            "circle",
+            "ring",
+            "square",
+            "diamond",
+            "plus",
+            "cross",
+            "asterix",
+            "triangle_up",
+            "triangle_down",
+            "triangle_left",
+            "triangle_right",
+            "heart",
+            "spade",
+            "club",
+            "pin",
+        ]
+        alt_names = {
+            "o": "circle",
+            "s": "square",
+            "D": "diamond",
+            "+": "plus",
+            "x": "cross",
+            "^": "triangle_up",
+            "<": "triangle_left",
+            ">": "triangle_right",
+            "v": "triangle_down",
+            "♥": "heart",
```

Comment:
emojis in the source code, very modern 😂

Reply:
Just unicode :)

Reply:
They're quite useful actually, as an example you can signal warnings with ⚠️ instead of using (complicated) shell text color output

Reply:
I had never thought of that!